### PR TITLE
Saul 210 - Option update

### PR DESCRIPTION
**Pull request by Renkon**

### CODE REVIEW

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py [hunk 1]
```diff
@@ -147,24 +147,14 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
-        c.argument('alt_key_x509',
-                   action='store_true',
-                   arg_group='Token Restriction Key Type',
-                   help='Use a certificate in PKCS12 format (X509Certificate2 in .NET) as an alternate token key.')
         c.argument('policy_option_name', help='The content key policy option name.')
         c.argument('policy_option_id', help='The content key policy option identifier.')
         c.argument('issuer', arg_group='Token Restriction', help='The token issuer.')
         c.argument('audience', arg_group='Token Restriction', help='The audience for the token.')
-        c.argument('token_key', arg_group='Token Restriction', help='Either a string (for symmetric key) or a filepath to a certificate (x509) or public key (rsa). Must be used with a Token Restriction Key Type flag.')
+        c.argument('token_key', arg_group='Token Restriction', help='Either a string (for symmetric key) or a filepath to a certificate (x509) or public key (rsa). To be used with token_key_type.')
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py]: `Must be used in conjunction with --token-key-type` wouldn't be better?. Same goes for alternate argument.

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py [hunk 2]
```diff
@@ -105,43 +105,87 @@ def remove_content_key_policy_option(client, resource_group_name, account_name,
...
+            policy_option.configuration.rental_duration = rental_duration
+    elif isinstance(policy_option.configuration, ContentKeyPolicyPlayReadyConfiguration):
+        if play_ready_template is not None and _valid_playready_configuration(play_ready_template):
+            _play_ready_configuration_factory(json.loads(play_ready_template))
 
+    return client.update(resource_group_name, account_name,
+                         content_key_policy_name, policy.options, policy.description)
+
+
+def update_content_key_policy_setter(client, resource_group_name, account_name, content_key_policy_name,
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py:187]: As we are not using setter now, we can remove it... just set setter name in commands.py with 'update' value to override the default setter value (create_or_update)

**Review by damatojuan — CHANGES_REQUESTED** [review]

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py (hunk 1 above)
Comment (Renkon) [reply src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py]: Fixed

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py (hunk 2 above)
Comment (Renkon) [reply src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py:187]: As previously discussed, it won't work if we don't use a custom setter.

Comment (damatojuan) [reply src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py:187]: Got it

**Review by damatojuan — APPROVED** [review]

**Review by NahuelBerg — APPROVED** [review]

**Pull request merged**